src/dialog_launcher.ts

```diff
@@ -110,10 +110,23 @@ export class Launcher extends search.Search {
             for (const [where, app] of this.desktop_apps) {
                 const name = app.name()
                 const name_match = name.toLowerCase()
-                const retain = name_match.startsWith(pattern)
-                    || name_match.includes(pattern)
-                    || levenshtein.compare(name_match, pattern) < 3
-
+                
+                let retain = false
+                if ( name_match.startsWith(pattern)
+                     || name_match.includes(pattern)
+                     || levenshtein.compare(name_match, pattern) < 3 {
+                    retain = true
+                } else if app.keywords() {
+                    for (const keyword of app.keywords()) {
+                        const keyword_match = keyword.toLowerCase()
+                        if ( keyword_match.startsWith(pattern)
+                             || keyword_match.includes(pattern)
+                             || levenshtein.compare(keyword_match, pattern) < 3) {
+                            retain = true
+                        }
+                    }
+                }
+                
```

Comment:
Maybe a cleaner solution would be to concatenate the name and keywords into an array, and then bail on the first condition of being true. Which could add the item to the list without needing a `retain` variable anymore.

```js
const name = app.name();
const keywords = app.keywords();
const app_items = keywords !== null ? keywords.concat(name) : [name];
for (const item of app_items) { }
```

Reply:
I basically did what you suggested, the only difference being that I concatenated keywords onto the name, instead of the other way around. Even though the sorting's not being done here, since we're breaking from the loop at the first match, it still seems better to compare the name before the potentially-long list of keywords.